### ISSUE TITLE
Data Service Generator: generated server does not compile when having a relationship

### ISSUE THREAD

**Issue by rburgst** [amplication/amplication#1131]
**Describe the bug**
Started a new project today, among other things the generated user class does not compile:

**To Reproduce**

```bash
yarn start
yarn run v1.22.10
$ nest start
...
          ~~~~~~~~~~
src/user/UserWhereInput.ts:52:14 - error TS2345: Argument of type '(text: string, reviver?: ((this: any, key: string, value: any) => any) | undefined) => any' is not assignable to parameter of type '(params: TransformFnParams) => any'.
  Types of parameters 'text' and 'params' are incompatible.
    Type 'TransformFnParams' is not assignable to type 'string'.

52   @Transform(JSON.parse)
                ~~~~~~~~~~
```

**Expected behavior**
downloaded server code can be built and run.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Mac OSX Catalina
 - Browser : N/A
 - Node Version 12.18.0

**Additional context**

The generated `UserWhereInput.ts`
```typescript
import { InputType, Field } from "@nestjs/graphql";
import { ApiProperty } from "@nestjs/swagger";
import { IsDate, IsOptional, IsString, ValidateNested } from "class-validator";
import { Type, Transform } from "class-transformer";
import { ClubWhereUniqueInput } from "../club/ClubWhereUniqueInput";
@InputType()
class UserWhereInput {
  @ApiProperty({
    required: false,
  })
  @IsDate()
  @Type(() => Date)
  @IsOptional()
  @Field(() => Date, {
    nullable: true,
  })
  createdAt?: Date;
  @ApiProperty({
    required: false,
    type: String,
  })
  @IsString()
  @IsOptional()
  @Field(() => String, {
    nullable: true,
  })
  firstName?: string | null;
  @ApiProperty({
    required: false,
    type: String,
  })
  @IsString()
  @IsOptional()
  @Field(() => String, {
    nullable: true,
  })
  id?: string;
  @ApiProperty({
    required: false,
    type: String,
  })
  @IsString()
  @IsOptional()
  @Field(() => String, {
    nullable: true,
  })
  lastName?: string | null;
  @ApiProperty({
    required: false,
    type: ClubWhereUniqueInput,
  })
  @Transform(JSON.parse)
  @ValidateNested()
  @Type(() => ClubWhereUniqueInput)
  @IsOptional()
  responsibleForClub?: ClubWhereUniqueInput | null;
  @ApiProperty({
    required: false,
  })
  @IsDate()
  @Type(() => Date)
  @IsOptional()
  @Field(() => Date, {
    nullable: true,
  })
  updatedAt?: Date;
  @ApiProperty({
    required: false,
    type: String,
  })
  @IsString()
  @IsOptional()
  @Field(() => String, {
    nullable: true,
  })
  username?: string;
}
export { UserWhereInput };
```

**Comment by rburgst** [comment]
note that the compiler error stems from the optional related entity `Club`

```
@Transform(JSON.parse)
  @ValidateNested()
  @Type(() => ClubWhereUniqueInput)
  @IsOptional()
  responsibleForClub?: ClubWhereUniqueInput | null;
```
basically I have compile errors wherever I have a related entity.

**Comment by yuval-hazaz** [comment]
resolved in #1080
